cirq/work/collector.py

```diff
@@ -103,7 +103,7 @@ def on_job_result(self, job: CircuitSampleJob,
         """
 
     def collect(self,
-                sampler: 'cirq.Sampler',
+                async_sampler: 'cirq.AsyncSampler',
```

Comment:
but now collector doesn't work for non-async samplers?  (I think this is why your build is not green?)

Reply:
Correct. But I think that's a necessary part of how Collector interacts with the world.

Reply:
Hm not being able use a Collector on Simulator seems bad (or I'm missing something)?